systemtest/produce_consume_basic_test.go

```diff
@@ -126,6 +193,33 @@ func assertProcessor(t testing.TB, assertions consumerAssertions) apmqueue.Proce
 	})
 }
 
+// sequenceConsumerAssertions asserts that records are consumed in sequence
+// assuming that all records are sent to same partitions. If records are
+// not sent to same partition then it will result in failures.
+type sequenceConsumerAssertions struct {
+	consumerAssertions
```

Comment:
seems we're only using the `records` field, should it just use `records *atomic.Int64`?

Reply:
Done